### ISSUE TITLE
Feature Request - Recognise ``vue`` files

### ISSUE THREAD

**Issue by Clemintina** [total-typescript/ts-reset#132]
Hello,

I am writing to request that ``ts-reset`` recognises ``vue`` files. From my testing, this amazing type utility only recognises ``ts`` and ``tsx`` files. 

I'm relatively new to Vue TS so if there is an existing method to have ``ts-reset`` support in vue files, it's always appreciated. 

Regards,
Kay

**Comment by mattpocock** [comment]
Thanks for the issue! Could you post some reproduction steps to show it not working in Vue?

**Comment by Clemintina** [comment]
Hello, 

I've forked a clean version of a Quasar project and added ts-reset as an example to emulate my project. 

[Code Sandbox](https://codesandbox.io/p/sandbox/ts-reset-vue-issue-with-quasar-oyclrx?file=%2Fsrc%2FApp.vue) - The example is in `App.vue`. 

I've used:
 `fetch` as my example of a promise which doesn't return `unknown`.
`JSON.parse` as my example of a the json parser returning `any`
`array.filter` as my example of the types still being `number | undefined`

My Project is utilising:

- Vue 3.2.47
- Quasar CLI 2.0.0
- TypeScript 5.0.2

I hope this is enough to help you reproduce the issue. Thank you for your time.

Regards,
Kay
